library/packages/src/lib/y2packager/repository.rb

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2020 SUSE LINUX GmbH, Nuremberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "yast/logger"
+
+Yast.import "Pkg"
+
+module Y2Packager
+  #
+  # This class represents a libzypp repository.
+  #
+  # @note The repository ID is dynamic and might be different
+  #   when the repositories are reloaded from disk.
+  #
+  # @since 4.2.54
+  class Repository
+    include Yast::Logger
+
+    attr_reader :alias, :name, :url, :raw_url, :id, :product_dir, :priority,
+      :enabled, :auto_refresh, :keep_packages, :service
+
+    def initialize(repo_alias: nil, auto_refresh: nil, enabled: nil, name: nil,
+      url: nil, raw_url: nil)
```

Comment:
why not having there id? and neither in find_all. There is reader, but nothing write it.